strawberryfields/devicespec.py

```diff
@@ -89,8 +89,10 @@ def gate_parameters(self) -> Mapping[str, Ranges]:
         >>> spec.gate_parameters
         {'squeezing_amplitude_0': x=0, x=1, 'phase_0': x=0, 0≤x≤6.283185307179586}
         """
-        gate_parameters = {}
+        if not self._spec["gate_parameters"]:
+            return None
```

Comment:
Is this the expected behaviour when the gate parameters is an empty dictionary? Or is it more correct to say
```suggestion
        if self._spec["gate_parameters"] is None:
            return None
```

Reply:
I'm not sure if any of the options would be considered more expected than the other, although it's probably safer to return empty dicts instead of `None` in this case.